### ISSUE TITLE
support for geojson.io

### ISSUE THREAD

**Issue by samsammurphy** [sat-utils/sat-search#78]
I was inspired by this [medium post](https://medium.com/devseed/sat-utils-cf8710e4521c) and excited to produce a beautiful search calendar..

`sat-search search --intersects aoi.geojson --datetime 2018-01-01/2018-03-30 -p "eo:cloud_cover<10" --save aoi-results.geojson --print-cal`

Obviously, I need my very own geojson file to run this search. My intuition tells me that _"[geojson.io](http://geojson.io) is a safe way to produce valid geojson, that will work for sure"_

However, I got this surprising error message

```
File "/home/sam/.virtualenvs/geo/lib/python3.6/site-packages/satsearch/search.py", line 75, in found
    return results['meta']['found']
KeyError: 'meta'
```
 
I looked into the dictionary from the offending line

https://github.com/sat-utils/sat-search/blob/d81e4774a41990b73b55db4b1e05b21062dd957c/satsearch/search.py#L75

using this code

        for key in results.keys():
            print('key =',key)
            print(results[key])

which told me that

```
key = code
500
key = description
Invalid GeoJSON Feature or geometry
```

This provided me the clue I needed! The server was expecting a feature or geometry. 

However, geojson.io saves to FeatureCollection. I manually extracted my single feature from the FeatureCollection and all worked out just fine. 

One way to provide better integration with geojson.io output might be to check for FeatureCollections with a single features in them and let that slide (i.e. silently extract the feature) but return an error message if a user passes a FeatureCollection with multiple features in it.

**Comment by samsammurphy** [comment]
p.s. I just realized that this issue actually touches any case where the users providers `Invalid GeoJSON Feature or geometry`. However, instead of that error message, they will receive the (not really v. helpful) message..

```
return results['meta']['found']
KeyError: 'meta'
```

Imo the API should return a 400 if it receives an malformed geojson but (from what I can tell) it seems to be returning a 200.

https://github.com/sat-utils/sat-search/blob/d81e4774a41990b73b55db4b1e05b21062dd957c/satsearch/search.py#L83

**Comment by matthewhanson** [comment]
Thanks for the issue @samsammurphy . Yes, there should at least be a better error message if the geometry is ill-formed.

But I also agree a FeatureCollection should also be supported, but I'm not sure how the best way is to do it. Should it use the first feature? Should it combine all the geometries together as a list of MultiPolygons (or MultiPoint or MultiLineString), in which case it should find any Items that intersect with any of the Features.

To make it general, I'd say that all Features should be appended to each other, although the common use case is for a single Feature I think.

**Comment by samsammurphy** [comment]
Good point @matthewhanson, I agree, appending Features together (if there are more than one) actually seems to be a better way to go.
